tracing/src/main/java/com/palantir/tracing/Tracers.java

```diff
@@ -195,6 +195,22 @@ public static Runnable wrap(String operation, Runnable delegate) {
         return new TracingAwareRunnable(Optional.of(operation), delegate);
     }
 
+    /**
+     * Wraps the given {@link Supplier} such that it uses the thread-local {@link Trace tracing state} at the time of
+     * it's construction during its {@link Supplier#get() execution}.
+     */
+    public static <V> Supplier<V> wrapSupplier(Supplier<V> delegate) {
+        return new TracingAwareSupplier<>(Optional.empty(), delegate);
+    }
```

Comment:
Most of the methods which don't take an operation name exist for historical reasons, ideally we'd require an operation name for new calls.

Reply:
Will remove.